### PR TITLE
Spotipy backend

### PR DESCRIPTION
**Pull request by nbehmane**
Changed how we were storing user data and integrated spotipy.

### CODE REVIEW

**Comment by nbehmane** [comment]
Failing travis since I didn't include sensitive data in the github

File: playlist.py [hunk 1]
```diff
@@ -3,6 +3,27 @@
...
+        self.name = spotify_playlist['name']
+        self.owner = spotify_playlist['owner']
+        self.public = spotify_playlist['public']
+        self.tracks = spotify_playlist['tracks']
+        self.uri = spotify_playlist['uri']
+        
+
+    
+
+"""
```
Comment (masonmcelvain) [inline playlist.py:26]: Are we commenting this out so you can save it for later? We could just delete it to reduce clutter. Seems like you could just go back in version control and get this if you needed it in the future.

**Review by masonmcelvain — APPROVED** [review]
Nice work! Minor comment about a commented out code block, but this all looks good to me

**Comment by scole02** [comment]
Looks good to me

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 install:
-  - pip install -r requirements.txt
+  - pip install flask
+  - pip install flask-cors
+  - pip install spotipy
+  - pip install dnspython
+  - pip install pymongo
+  - pip install python-dotenv
+  - pip install coverage
+  - pip install pytest
```
Comment (masonmcelvain) [inline .travis.yml:13]: if our requirements.txt is up to date, we could just list `pip install -r requirements.txt` and remove the rest of these

File: README.md [hunk 3]
```diff
@@ -8,7 +8,17 @@ Pair playlists and songs with locations!
...
+* Go inside `env/Scripts` and run the activate script for your OS
+  * Windows: `Activate.ps1`
+  * Linux / Mac: `activate.sh`
+* Install the following packages:
+  * `pip install flask flask-cors pymongo dnspython python-dotenv spotipy`
+* Create a file called `.env` and set the following environment variables:
+  * CLIENT_ID, CLIENT_SECRET, MONGO_URL, REDIRECT_URI
+* run `flask run`
 
 ## Database
```
Comment (masonmcelvain) [inline README.md:23]: We should delete the parts of the README that aren't associated with the backend (ie Database, Frontend parts)

File: app.py [hunk 4]
```diff
@@ -1,56 +1,243 @@
...
 
+scope = [
+        'playlist-read-private',
+            'playlist-modify-public',
+            'playlist-modify-private',
+            'user-library-read',
+            'user-library-modify',
+            'user-top-read',
+            'user-read-email'
+            ]
```
Comment (masonmcelvain) [inline app.py:30]: indentation is funky

File: app.py [hunk 5]
```diff
@@ -1,56 +1,243 @@
...
-            response = [playlist.getData() for playlist in db_playlists]
-            return jsonify(response)
+    db_playlists = p_repo.read()
+    data = request.json # Get the location of the user
+    radius = data['radius']
+    if request.method == "POST":
+        close_playlists = []
+        for playlist in db_playlists:
+            distance = math.sqrt(((playlist['location'][0] - data['location'][0]) ** 2) + 
+                       ((playlist['location'][1] - data['location'][1]) ** 2))
```
Comment (masonmcelvain) [inline app.py:228]: this is dense. maybe we could extract this out into it's own function with a descriptive name

File: backend_tests.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+from dotenv import load_dotenv
+import os
+load_dotenv()
+
+
+'''TODO:
+    1. Test create/update/read/remove for Playlist
+    2. Test create/update/read/remove for GeoMusicUser
+    3. Test pulling user from spotify (user information and playlists)
+        and adding that user to our database'''
```
Comment (masonmcelvain) [inline backend_tests.py:20]: is this still a valid TODO, or can it be deleted?

File: backend_tests.py [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+scope = [
+        'playlist-read-private',
+            'playlist-modify-public',
+            'playlist-modify-private',
+            'user-library-read',
+            'user-library-modify',
+            'user-top-read',
+            'user-read-email'
+            ]
```
Comment (masonmcelvain) [inline backend_tests.py:32]: indentation

**Review by masonmcelvain — CHANGES_REQUESTED** [review]
Nice job! This is solid. A few minor comments, mostly around documentation / styling

**Comment by masonmcelvain** [comment]
We'll also need to settle the merge conflict once the README is ready to go.

File: .travis.yml (hunk 2 above)
Comment (nbehmane) [reply .travis.yml:13]: For sure, I'll make sure remove those from the README and just include it in requirements.txt

File: README.md (hunk 3 above)
Comment (nbehmane) [reply README.md:23]: For sure, I think the database steps should be there just for the backend as a whole to work properly. Do you think I should still get rid of it in the readme?

File: app.py (hunk 4 above)
Comment (nbehmane) [reply app.py:30]: Will fix those indentation issues.

File: app.py (hunk 5 above)
Comment (nbehmane) [reply app.py:228]: I agree with you, that should be refactored out of there. I'll update the code and push the changes.

File: backend_tests.py (hunk 6 above)
Comment (nbehmane) [reply backend_tests.py:20]: These are not valid TODO's anymore, will get rid of them in the code. Thanks for pointing that out.

File: README.md (hunk 3 above)
Comment (masonmcelvain) [reply README.md:23]: Oh, yea keep the database stuff sorry!

**Review by masonmcelvain — APPROVED** [review]
Nice, thanks!

**Review by kylene-phillips — COMMENTED** [review]
Great changes and important additions. Thanks for making the sensitive data hidden now. Been working with the backend and your code does everything except update the images in the backend which we need. We will work on that together

**Review by kylene-phillips — COMMENTED** [review]
Looks good, thank you!

**Pull request merged**